sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -210,8 +206,13 @@ object Cast {
     case _ => false  // overflow
   }
 
+  /**
+   * Returns `true` iif it should change the nullability of this type in the container,
+   * e.g. map type, array type, expression, etc. This function should take the nullability
+   * in the container into account. For example, it should not force to a nullable type
+   * when null type is non-nullable, which means an empty array of null type.
```

Comment:
nit: `iif` -> `iff`

Reply:
Actually, I used the term `container` intentionally. There's one more place that's used with expressions' nullability at `Cast`.

```scala
  override def nullable: Boolean = Cast.forceNullable(child.dataType, dataType) || child.nullable
```

 Let me try to make it better a bit while I am here.